Update go libraries page with new intro (#3826)
open-telemetry/opentelemetry.io: content/en/docs/languages/go/libraries.md

weight: 40
---

{{% docs/languages/libraries-intro Go %}}

> ## Use Instrumentation Libraries

Reviewer: ```suggestion
## Use instrumentation libraries
```

If a library does not come with OpenTelemetry out of the box, you can use
[instrumentation libraries](/docs/specs/otel/glossary/#instrumentation-library)

> in order to generate telemetry data for a library or framework.

Reviewer 2: ```suggestion
to generate telemetry data for a library or framework.
```

Shorter is always better!

> For example,
> [the instrumentation library for `net/http`](https://pkg.go.dev/go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp)

Reviewer: ```suggestion
For example, the
[instrumentation library for `net/http`](https://pkg.go.dev/go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp)
```

> will automatically create [spans](/docs/concepts/signals/traces/#spans) and

Reviewer 2: ```suggestion
automatically creates [spans](/docs/concepts/signals/traces/#spans) and
```

> metrics(/docs/concepts/signals/metrics) based on the HTTP requests.

Reviewer: Did you mean to make a link here?

```suggestion
[metrics](/docs/concepts/signals/metrics/) based on the HTTP requests.
```

Instrumentation libraries can do things like generate telemetry data for inbound
and outbound HTTP requests, but they don't instrument your actual application.

> To get richer telemetry data, add your own
> [instrumentation](../instrumentation/) to your code to enrich your telemetry
> data from instrumentation libraries with instrumentation from your running
> application.

Reviewer: Hmm, how about:

```suggestion
Enrich your telemetry data by integrating [custom instrumentation](../instrumentation/) into your code. This supplements the standard library telemetry, and can offer deeper insights into your running application.
```
